mne/decoding/time_delaying_ridge.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+"""TimeDelayingRidge class."""
+# Authors: Eric Larson <larson.eric.d@gmail.com>
+#          Ross Maddox <ross.maddox@rochester.edu>
+#
+# License: BSD (3-clause)
+
+import warnings
+
+import numpy as np
+
+from .base import BaseEstimator
+from ..filter import next_fast_len
+
+
+def _compute_corrs(X, y, smin, smax, fit_intercept):
+    """Compute the auto- and cross-correlations."""
+    len_trf = smax - smin
+    len_x, n_ch_x = X.shape
+    len_y, n_ch_y = y.shape
+    assert len_x == len_y
+
+    if fit_intercept:
+        # We could do this in the Fourier domain, too, but it should
+        # be a bit cleaner numerically do do it here.
+        X = X - np.mean(X, axis=0, keepdims=True)
+        y = y - np.mean(y, axis=0, keepdims=True)
+    n_fft = next_fast_len(X.shape[0] + max(smax, 0) - min(smin, 0) - 1)
+    X_fft = np.fft.rfft(X.T, n_fft)
+    y_fft = np.fft.rfft(y.T, n_fft)
+    # del X, y
+
+    # compute the autocorrelations
+    ac = np.zeros((n_ch_x, n_ch_x, len_trf * 2 - 1))
+    for ch0 in range(n_ch_x):
+        for ch1 in range(ch0, n_ch_x):
+            # This is equivalent to:
+            # ac_temp = np.correlate(X[:, ch0], X[:, ch1], mode='full')
+            # ac_temp = np.roll(ac_temp_2, X.shape[0])
+            ac_temp = np.fft.irfft(X_fft[ch0] * X_fft[ch1].conj(), n_fft)
```

Comment:
ugh yes I hadn't even seen `.conj()`. (I think for numpy arrays `.imag` and `.real` are strided views on the array, so that could help - but you might feel like wanting to avoid writing out complex mult yourself)